### DISCUSSION TITLE
Integration with Spring Security

### DISCUSSION THREAD

**Discussion by victorlevasseur** [Netflix/dgs-framework#57] [Q&A] [answered]
Hello,

According to this page https://netflix.github.io/dgs/advanced/security/ there's some integration between DGS and Spring Security. Is it making sure the SecurityContextHolder is properly configured to be available in all the datafetchers and dataloaders despite having an async query and mutation strategy ?

**Comment by paulbakker** [comment, marked as answer]
The Spring Security integration lets you use `@Secured` on data fetchers. It does **not** take care of SecurityContextHolder. 

At Netflix we have an executor that is set up with thread local data, and for DGS we use this executor when creating CompletableFutures. You could possibly use a similar pattern, but it doesn't come out of the box with DGS.

**Reply by eneoli** [reply]
@paulbakker Could you please elaborate how you did setup such an executor? I face a similar issue and tried that approach already - without success.
I really would appreciate a hint.

**Reply by srinivasankavitha** [reply]
The docs here provide a good overview of the options available: https://docs.spring.io/spring-security/reference/features/integrations/concurrency.html, specifically setting up a `delegatingsecuritycontextasynctaskexecutor`

More examples here as well:
https://howtodoinjava.com/spring-security/spring-security-context-propagation/#32-using-delegatingsecuritycontextasynctaskexecutor

Hope this helps.
